code/modules/mob/living/login.dm

```diff
@@ -26,3 +26,5 @@
 		changeling.regain_powers()
 
 	med_hud_set_status()
+
+	update_fov_client()
```

Comment:
This is still hitting my concerns of increasing surface area. The FOV component should be listening for the login signal.

Reply:
It cant listen for Login because it won't exist at that time

Reply:
So is this just being added to every mob, regardless of if it is actually being used (whether through items or config)? It feels like it would already exist if it was actually needed.

Reply:
The component is only added to the mob if the mob has a field of view and a client. `update_fov_client()` is a proc to check the conditions and update/create the component if the mob is eligible.